qutebrowser/utils/jinja.py

```diff
@@ -127,3 +130,33 @@ def render(template, **kwargs):
 
 environment = Environment()
 js_environment = jinja2.Environment(loader=Loader('javascript'))
+
+
+@functools.lru_cache()
+def template_config_variables(template: str) -> typing.FrozenSet[str]:
+    """Return a frozenset of config variables that is used in template."""
+    pending = [environment.parse(template)]  # a list of unvisited nodes
+    result = []  # type: typing.List[str]
```

Comment:
Why not make this a set and return it directly, instead of making it a list and converting it to a frozenset?

Reply:
May break lru cache if something tamper with the result.

Reply:
Fair point, but then this should be a Set instead - there's no point in it being a list, we don't care about duplicates or order.

Reply:
I don't think it matters. We don't need any set functionalities in this case (only add an element and iterate)